apps/meteor/server/models/raw/Users.js

```diff
@@ -270,18 +262,115 @@ export class UsersRaw extends BaseRaw {
 		return this.findPaginated(query, options);
 	}
 
+	findPaginatedActiveUsersByRoomIdWithHighestRole(
+		searchTerm,
+		rid,
+		searchFields,
+		options = {},
+		extraQuery = [],
+		{ startsWith = false, endsWith = false } = {},
+	) {
+		const regexString = (startsWith ? '^' : '') + escapeRegExp(searchTerm) + (endsWith ? '$' : '');
```

Comment:
The regexp match could be problematic perf-wise, generally is better to always check for the match for the start of the string so the index could cover it, but at least on the usage on this PR, the option is not passed so it will try to match without index. 

Depending on the amount of `search fields` and the number of users, this will be slow.

Reply:
idk if that kind of search is a requirement, so I'm just sharing the thought on here.

Reply:
This new endpoint will replace the current `.members` endpoints in the UI (we'll use it to display the members lists in rooms). So it should have the same capabilities as `.members`, which means this is a requirement for now :/